### PR TITLE
TMEDIA 252 nav list item transparency issue

### PR DESCRIPTION
**Pull request by JackHowa**
## Description
When the sections menu has more items than can fit within the viewport, and the sections menu is closed additional white space is added to the bottom of the page.

## Jira Ticket
- [TMEDIA-252](https://arcpublishing.atlassian.net/browse/TMEDIA-252)

## Acceptance Criteria
Expected Behavior
1. No additional scrolling should happen when you reach the bottom of a page when
a. The sections menu is closed
b. The sections menu is closed and have more items than would fit within the viewport
2. When the sections menu is open
a. The sections menu should scroll to allow a user to see all menu items
b. User should be able to open multiple sub sections and still scroll to see all menu items

## Test Steps
- Add test steps a reviewer must complete to test this PR

1. Checkout this branch `git checkout TMEDIA-252-nav-list-whitespace-issue`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/shared-styles,@wpmedia/header-nav-chain-block`
3. Go to http://localhost/homepage/?_website=the-gazette 
4. Click sidebar section
5. Open subsection
6. Scroll down the page and ensure background color of items is not transparent

### Before
Screen Shot 2021-07-27 at 11 58 00

### After
Screen Shot 2021-07-27 at 11 57 25

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [ ] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

**Comment by nschubach** [comment]
Is this supposed to cover the issue of cut off content on the menu too?

image

image

Seems like it in AC:
> b. User should be able to open multiple sub sections and still scroll to see all menu items

**Comment by nschubach** [comment]
I briefly looked into that and since we are setting the height to 100vh then shifting the margin down, the content is all showing, but it's shifted down by the size of the header bar.  The way I got it to show was to do the following based on the margin:
image
But the header size changes, so I don't know if that's appropriate here

**Comment by JackHowa** [comment]
hmm interesting. I can look into that cutoff issue. seems near in scope at least @nschubach 
Screen Shot 2021-07-27 at 13 22 07

**Comment by JackHowa** [comment]
updated -- big thanks @nschubach 

# before 
- last items were being hidden (brand studio) or unstyled on scroll
image
Screen Shot 2021-07-27 at 13 57 23

# after

## desktop 

- accommodating on desktop and mobile for margins and items above with hard-coded values 

Screen Shot 2021-07-27 at 13 58 54

## mobile 

Screen Shot 2021-07-27 at 13 51 17
Screen Shot 2021-07-27 at 13 51 10

File: blocks/shared-styles/scss/_header-nav.scss [hunk 1]
```diff
@@ -408,17 +409,27 @@ body.nav-open {
   }
 
   &.open {
+    // dimmed background
     background: rgba(25, 25, 25, 0.5);
-    height: 100vh;
+    // total viewport height - header height
+    height: calc(100vh - 100px);
```
Comment (matthewroach) [inline blocks/shared-styles/scss/_header-nav.scss]: What's the `100px` value? This is a magic number as the header height is controllable as a custom field

File: blocks/shared-styles/scss/_header-nav.scss [hunk 2]
```diff
@@ -408,17 +409,27 @@ body.nav-open {
...
     overflow-y: scroll;
     transform: translate(0, 0);
     width: 100%;
 
     .section-menu {
-      height: 100vh;
+      // total viewport height - header height - top margin style
+      height: calc(100vh - 100px - 13px);
```
Comment (matthewroach) [inline blocks/shared-styles/scss/_header-nav.scss]: Again, magic numbers are a little scary with possible changes

File: blocks/shared-styles/scss/_header-nav.scss [hunk 3]
```diff
@@ -408,17 +409,27 @@ body.nav-open {
...
     }
 
     @media screen and (max-width: map-get($grid-breakpoints, 'md')) {
+      // viewport height - top nav height
+      height: calc(100vh - 56px);
+
+      .section-menu {
+        // viewport height - search sub nav height - top nav height
+        height: calc(100vh - 48px - 56px);
```
Comment (matthewroach) [inline blocks/shared-styles/scss/_header-nav.scss]: More magic numbers?

**Review by matthewroach — CHANGES_REQUESTED** [review]
Seeing a lot of white space at the bottom of the page in FireFox still

firefox-white-space-end-of-page

**Comment by JackHowa** [comment]
ok interesting I'll have to maybe look at refactoring this more @matthewroach

File: blocks/shared-styles/scss/_header-nav.scss (hunk 1 above)
Comment (nschubach) [reply blocks/shared-styles/scss/_header-nav.scss]: That 100px is the margin that is applied to the top of the menu and is basically the header height.  Unfortunately, it's not easy to expose that to the stylesheet where that is needed I don't think.  I have been playing around with options to counter this but have no real solutions that don't up-end the page a bit more than comfort allows.

**Comment by JackHowa** [comment]
I removed the hard-coded widths and heights from the stylesheet and am doing it dynamically based on custom fields. So this is an improvement; I did not realize that the height of the nav was variable. that's my bad. I did remove the top margin 13 px that seemed like it was for style and did not want to hard-code into the styled component.  thank you @matthewroach 

with scroll down 

Screen Shot 2021-07-27 at 16 47 13

without scroll down 

Screen Shot 2021-07-27 at 16 47 37

**Comment by JackHowa** [comment]
The firefox bug seems to be common based on other reports and their bugtracker. the overflow property does not seem to be working as it does in chromium browsers 

Screen Shot 2021-07-27 at 16 17 13


https://stackoverflow.com/questions/28636832/firefox-overflow-y-not-working-with-nested-flexbox/28639686#28639686
Screen Shot 2021-07-27 at 16 49 35


I think this solves the bug I reported of the bottom list items not being able to be styled as expected. I'm not too sure what our support for firefox should be here.

File: blocks/shared-styles/scss/_header-nav.scss (hunk 3 above)
Comment (JackHowa) [reply blocks/shared-styles/scss/_header-nav.scss]: updated -- that's my bad

File: blocks/shared-styles/scss/_header-nav.scss (hunk 1 above)
Comment (JackHowa) [reply blocks/shared-styles/scss/_header-nav.scss]: I've updated this for styled components and custom fields

**Comment by matthewroach** [comment]
@JackHowa - The original report was due based on seeing the issue in Firefox, and we have had reports and fixed related issues in the past that have affected on FireFox - https://arcpublishing.atlassian.net/browse/TMEDIA-174.
The way the section menu works, does have some quirks cross browser in it's current implementation which we noted during grooming - it's very possible it needs refactored to a better set up as a whole

**Comment by nschubach** [comment]
I just pulled down the latest and ran it (changed the run command to add the header block: `npx fusion start -f -l @wpmedia/engine-theme-sdk,@wpmedia/header-nav-chain-block`) and it's now doing this in Chrome:

image

I'm with Matt here.  There is a structure issue here.

**Comment by JackHowa** [comment]
@matthewroach updated per firefox fix 


before 

- scroll down the page on firefox https://corecomponents-the-gazette-prod.cdn.arcpublishing.com/ 

Screen Shot 2021-07-28 at 11 00 32


after

- go to  http://localhost/homepage/?_website=the-gazette
- scroll down the page. validate that it doesn't continue scrolling past the footer 
- play with the sidebar nav. ensure that expanding and closing sites does overflow and make the page longer

Screen Shot 2021-07-28 at 10 53 46

**Comment by JackHowa** [comment]
@matthewroach updated per firefox fix -> good to go


before 

- scroll down the page on firefox https://corecomponents-the-gazette-prod.cdn.arcpublishing.com/ 

Screen Shot 2021-07-28 at 11 00 32


after

- go to  http://localhost/homepage/?_website=the-gazette
- scroll down the page. validate that it doesn't continue scrolling past the footer 
- play with the sidebar nav. ensure that expanding and closing sites does overflow and make the page longer

Screen Shot 2021-07-28 at 10 53 46

**Comment by JackHowa** [comment]
@matthewroach added back the top-margin that we talked about. 

- I've been testing with my actual phone and the simulators and haven't come to a good solution yet for the bottom margin
- I forgot the safe area inset doesn't seem accommodate for the actual safari bottom toolbar. but I've had some success with the margin-bottom on the nav-section for mobile breakpoints

**Comment by JackHowa** [comment]
Ensure bottom link "Brand Studio" (on the gazette) is always visible after scrolling 

# before 

## desktop 

Screen Shot 2021-07-29 at 10 10 45

## mobile 

Screen Shot 2021-07-29 at 10 10 15

# after

## desktop 

Screen Shot 2021-07-29 at 10 07 48

## mobile 

Screen Shot 2021-07-29 at 10 08 57

**Comment by nschubach** [comment]
http://localhost/pf/homepage/?_website=the-gazette

I'm still getting this weirdness when making the window really short and scrolling down in the menu:
(Window is ~500px tall, menu is 760, missing content is roughly the header height I think)
image
(This is scrolled as far down as I can get)

`flex-basis` helps with the background:
image

but the content is not scrolling to the end yet... still investigating.  (extremely short windows may also just be an exception or another issue to track... I think that would be ok.)

**Comment by JackHowa** [comment]
I'm not able to replicate this @nschubach 

what blocks are you linking? should be: 

`npx fusion start -f -l @wpmedia/shared-styles,@wpmedia/header-nav-chain-block`

Also I'd make sure that the branch is updated. A lot has changed since this was started

Screen Shot 2021-07-29 at 15 26 33

**Comment by nschubach** [comment]
I had the wrong link 🤦 .  looks good now.

**Review by nschubach — APPROVED** [review]

**Pull request merged**